collector/cpu.go

```diff
@@ -3,23 +3,54 @@
 package collector
 
 import (
+	"strconv"
 	"strings"
 
 	"github.com/StackExchange/wmi"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/common/log"
+	"golang.org/x/sys/windows/registry"
 )
 
 func init() {
 	Factories["cpu"] = NewCPUCollector
 }
 
+//A function to get windows version from registry
+
+func getWindowsVersion() float64 {
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\Windows NT\CurrentVersion`, registry.QUERY_VALUE)
+	if err != nil {
+		log.Warn("Couldn't open registry", err)
+		return currentv_flt
```

Comment:
I don't understand this, looks like this variable isn't defined yet here? How does this work?
Could we just return 0 on any error? Or maybe NaN. The only usage is comparing with the magic constant 6.05, right?

Reply:
I tried building the PR locally, and get errors about the undefined variable, so I'm guessing this is also why CI fails @martinlindhe. Really weird that we don't get any build errors in the output, though...

Reply:
Yes, you are correct, I will return 0 on error. That should failback to legacy wmi query for cpu metrics. Thanks for catch!